tsfile/src/main/java/org/apache/iotdb/tsfile/write/record/Tablet.java

```diff
@@ -138,7 +138,7 @@ private void addValueOfDataType(
       case TEXT:
         {
           Binary[] sensor = (Binary[]) values[indexOfSchema];
-          sensor[rowIndex] = value != null ? (Binary) value : Binary.EMPTY_VALUE;
+          sensor[rowIndex] = value != null ? new Binary((String) value) : Binary.EMPTY_VALUE;
```

Comment:
Thanks for your contribution!

IMO, we should also support adding a Binary value here to keep the compatibility. :)

Reply:
Besides, consider adding some tests in `IoTDBSessionSimpleIT`?

Reply:
Thanks for your valuable advice! I will work on them immediately.